src/storage/src/sink/kafka.rs

```diff
@@ -1169,7 +1175,7 @@ where
 
                     if let Some(gate) = latest_ts {
                         assert!(
-                            as_of.frontier.iter().all(|ts| *ts <= gate),
+                            PartialOrder::less_equal(&as_of.frontier, &Antichain::from_elem(gate)),
```

Comment:
This is equivalent but more principled, right?

Reply:
that's right! I made the change so it's parallel to the change above

Reply:
I think it might not be strictly equivalent, but also correct here. `PartialOrder::less_equals(as_of, gate)` says "for all elements g of gate, there must be an element a of as_of where a <= g".

What we had here previously was saying "for all elements a of as_of it must be that a <= gate"